File: core/src/bara/game/components/LayerComponent.java
package bara.game.components;

import com.badlogic.ashley.core.Component;
import com.badlogic.gdx.utils.Pool;

public class LayerComponent implements Component, Pool.Poolable {
    public int layer = 0;

    @Override
    public void reset() {
        layer = 0;
    }
}

File: core/src/bara/game/components/LayerComponent.kt
package bara.game.components

import bara.game.ashley.GameComponent
import com.badlogic.ashley.core.Component
import com.badlogic.gdx.utils.Pool
import ktx.ashley.EngineEntity

class LayerComponent : GameComponent() {
    var layer = 0

    override fun reset() {
        layer = 0
    }

    override fun buildOnto(engineEntity: EngineEntity) {
        val tempComponent = this
        engineEntity.with<LayerComponent> {
            layer = tempComponent.layer
        }
    }
}
